src/fetching/provider/query_service.rs

```diff
@@ -677,6 +679,62 @@ mod test {
         );
     }
 
+    #[async_std::test]
+    async fn test_pruning() {
```

Comment:
I don't think this test is really specific to fetching at all, unlike the other tests in this file. Consider moving this to `data_source/sql.rs`? Really it should be a generic test, but I guess it won't work for the file system since we haven't implemented pruning there

Reply:
deleted this test. There are separate tests for pruning in sql.rs